### ISSUE TITLE
Support distributed tracing across Pub/Sub

### ISSUE THREAD

**Issue by EricStG** [googleapis/google-cloud-dotnet#9157]
Thanks for stopping by to let us know something could be better!

**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console]I want to be able to correlate requests across services made through pub/sub

 **Describe the solution you'd like**
Ideally, support for OpenTelemetry that will:
- Create a new activity when sending/receiving messages
- Enrich pub/sub messages with the attributes for correlation / baggage propagation
- 

 **Describe alternatives you've considered**
We're doing it manually for now with the OpenTelemetry propagators.

 **Additional context**
Similar to #8366, but we haven't had issues with automated instrumentation over HTTP, but pub/sub is a big blindspot for us.
Sounds like there are internal discussions around supporting this over different languages like [golang](https://github.com/googleapis/google-cloud-go/issues/4665) and [nodejs](https://github.com/googleapis/nodejs-pubsub/issues/1389), but can't tell how far those discussions have gone

**Comment by jskeet** [comment]
Assigning to Amanda for comment as she knows more about OpenTelemetry than I do. (But fundamentally I wouldn't expect to see anything here *soon*.)

**Comment by amanda-tarafa** [comment]
@EricStG Yes, there are ongoing discussion to support OpenTelemetry for all libraries, inlcuding Pub/Sub, across all languages, including .NET, but those are on very early stages. As @jskeet said, don't expect anything soon.

It seems you already have implemented tracing manually, so this might no be useful for you, but you can use [Google.Cloud.Diagnostics.Common](https://cloud.google.com/dotnet/docs/reference/Google.Cloud.Diagnostics.Common/latest) to add tracing to non-HTTP contexts. Apart from the linked docs, see #5897 and #6367 for more context. And of course, create specific issues if you have questions or run into problems.

If it's OK I'd prefer to close this issue as a duplicate of #8366 as we are including Pub/Sub in all our explorations.
